File: fsharp/bob/Bob.fs
module Bob

let (|Silence|Shouting|Question|Statement|) (text: string) =
    if text.Trim() = "" then
        Silence
    else if text = text.ToUpper() && text <> text.ToLower() then 
        Shouting
    else if text.EndsWith("?") then
        Question
    else
        Statement

let hey (text) =
    match text with
    | Silence -> "Fine. Be that way!"
    | Shouting -> "Whoa, chill out!"
    | Question -> "Sure."
    | Statement -> "Whatever."

File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn yelling? [input]
  (and (= (str/upper-case input) input)
       (not= (str/lower-case input) input)))

(defn response-for [input] ;; <- arglist goes here
  (cond 
    (yelling? input) "Whoa, chill out!" 
    (str/ends-with? input "?") "Sure." 
    (str/blank? input) "Fine. Be that way!"
    :else "Whatever."))

